contracts/tokenbridge/libraries/gateway/GatewayRouter.sol

```diff
@@ -144,4 +147,77 @@ abstract contract GatewayRouter is TokenGateway, IGatewayRouter {
         }
         return TokenGateway(gateway).calculateL2TokenAddress(l1ERC20);
     }
+
+    /**
+     * @notice Bridge ERC20 token using the registered or otherwise default gateway with standard EIP 2612 call to permit. 
+                Compatible with older gateways without OutboundTransferCustomRefund
+     * @notice Safe from reentrancy as there are no calls in the function into the caller's address
+     * @param _token L1 address of ERC20
+     * @param _to Account to be credited with the tokens in the L2 (can be the user's L2 account or a contract), not subject to L2 aliasing
+                  This account, or its L2 alias if it have code in L1, will also be able to cancel the retryable ticket and receive callvalue refund
+     * @param _amount Token Amount
+     * @param _maxGas Max gas deducted from user's L2 balance to cover L2 execution
+     * @param _gasPriceBid Gas price for L2 execution
+     * @param _data encoded data from router and user
+     * @param _permitData signature and deadline params of permit
+    */
+    function outboundTransferWithEip2612Permit(
+        address _token,
+        address _to,
+        uint256 _amount,
+        uint256 _maxGas,
+        uint256 _gasPriceBid,
+        bytes calldata _data,
+        PermitData calldata _permitData
+    ) public payable virtual override returns (bytes memory) {
+        address gateway = getGateway(_token);
```

Comment:
Actually we can also have the gateway as 1 of the external function parameter, since the UI need to figure out which gatway to sign permit anyway.